diff --color -Naur a/meson.build b/meson.build
--- a/meson.build	2022-03-22 12:12:57.000000000 -0400
+++ b/meson.build	2022-05-27 12:43:42.431230376 -0400
@@ -55,7 +55,7 @@
 stability_version_number = (major_version != 0 ? minor_version : micro_version)
 stable = (stability_version_number % 2 == 0)
 
-conf.set10('BABL_UNSTABLE', not stable, Description:
+conf.set10('BABL_UNSTABLE', not stable, description:
   'Define to 1 if this is an unstable version of BABL.')
 
 conf.set       ('BABL_MAJOR_VERSION',    '@0@'.format(major_version))
