diff -ruN a/Source/LibRawLite/internal/dcraw_common.cpp b/Source/LibRawLite/internal/dcraw_common.cpp
--- a/Source/LibRawLite/internal/dcraw_common.cpp	2022-04-09 19:06:06.993197060 -0400
+++ b/Source/LibRawLite/internal/dcraw_common.cpp	2022-04-09 19:06:43.677168902 -0400
@@ -9357,7 +9357,7 @@
   strncpy (th->t_desc, desc, 512);
   strncpy (th->t_make, make, 64);
   strncpy (th->t_model, model, 64);
-  strcpy (th->soft, "dcraw v"DCRAW_VERSION);
+  strcpy (th->soft, "dcraw v" DCRAW_VERSION);
   t = localtime (&timestamp);
   sprintf (th->date, "%04d:%02d:%02d %02d:%02d:%02d",
       t->tm_year+1900,t->tm_mon+1,t->tm_mday,t->tm_hour,t->tm_min,t->tm_sec);
