diff --color -Naur a/gexiv2/meson.build b/gexiv2/meson.build
--- a/gexiv2/meson.build	2018-11-18 04:37:00.000000000 -0500
+++ b/gexiv2/meson.build	2022-05-27 13:45:03.963930201 -0400
@@ -33,14 +33,6 @@
 
 install_headers(gexiv2_headers, subdir : 'gexiv2')
 
-# Static map file
-mapfile = 'gexiv2.map'
-if host_machine.system() == 'darwin'
-    vflag = ''
-else
-    vflag = '-Wl,--version-script,@0@/@1@'.format(meson.current_source_dir(), mapfile)
-endif
-
 gexiv2 = library('gexiv2',
                  ['gexiv2-metadata.cpp',
                   'gexiv2-metadata-exif.cpp',
@@ -60,7 +52,6 @@
                  gexiv2_headers +
                  [version_header] +
                  enum_sources,
-                 link_args : vflag,
                  include_directories : include_directories('..'),
                  version: '2.0.0',
                  dependencies : [gobject, exiv2],
