diff -ruN a/glib/gspawn.c b/glib/gspawn.c
--- a/glib/gspawn.c	2022-04-08 12:42:11.803752651 -0400
+++ b/glib/gspawn.c	2022-04-08 12:42:36.007753960 -0400
@@ -1339,7 +1339,7 @@
    *
    * Handle ENOSYS in case it’s supported in libc but not the kernel; if so,
    * fall back to safe_fdwalk(). */
-  if (close_range (lowfd, G_MAXUINT) != 0 && errno == ENOSYS)
+  if (close_range (lowfd, G_MAXUINT, 0) != 0 && errno == ENOSYS)
 #endif  /* HAVE_CLOSE_RANGE */
   (void) safe_fdwalk (close_func, GINT_TO_POINTER (lowfd));
 #endif
