diff --color -Naur a/configure b/configure
--- a/configure	2022-07-31 05:20:45.583028238 -0400
+++ b/configure	2022-07-31 05:21:17.422924550 -0400
@@ -8786,8 +8786,8 @@
 MATH_LIB=-lm
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for native Win32" >&5
 $as_echo_n "checking for native Win32... " >&6; }
-LIB_EXE_MACHINE_FLAG=X86
-EXE_MANIFEST_ARCHITECTURE=X86
+LIB_EXE_MACHINE_FLAG=ARM
+EXE_MANIFEST_ARCHITECTURE=ARM
 DISABLE_ON_W32=''
 case "$host" in
   *-*-mingw*)