diff --git a/src/util/xmlconfig.c b/src/util/xmlconfig.c
index 26a811b..c8a2d4e 100644
--- a/src/util/xmlconfig.c
+++ b/src/util/xmlconfig.c
@@ -56,7 +56,6 @@ static inline void regfree(regex_t* r) {}
 #endif
 #include <fcntl.h>
 #include <math.h>
-#include "strndup.h"
 #include "u_process.h"
 #include "os_file.h"

@@ -237,7 +236,7 @@ parseValue(driOptionValue *v, driOptionType type, const char *string)
       break;
    case DRI_STRING:
       free(v->_string);
-      v->_string = strndup(string, STRING_CONF_MAXLEN);
+      v->_string = strdup(string);
       return true;
    case DRI_SECTION:
       unreachable("shouldn't be parsing values in section declarations");
