diff '--color=auto' -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-07-01 00:16:24.720833934 -0400
+++ b/CMakeLists.txt	2025-07-01 00:58:09.480694340 -0400
@@ -47,6 +47,11 @@
     set(COMPILER_IS_GCC_OR_CLANG ON)
 endif ()
 
+if (CMAKE_CXX_COMPILER MATCHES "armv7-w64-mingw32")
+	set(COMPILER_IS_ARM_MINGW ON)
+	message(STATUS "Using an ARMv7 MinGW compiler...")
+endif ()
+
 # -----------------------------------------------------------------------------
 # Determine the target processor
 # -----------------------------------------------------------------------------
diff '--color=auto' -Naur a/Source/cmake/OptionsQt.cmake b/Source/cmake/OptionsQt.cmake
--- a/Source/cmake/OptionsQt.cmake	2025-07-01 00:16:24.631399331 -0400
+++ b/Source/cmake/OptionsQt.cmake	2025-07-01 00:43:24.564280802 -0400
@@ -721,7 +721,7 @@
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-expansion-to-defined")
 endif ()
 
-if (WIN32 AND COMPILER_IS_GCC_OR_CLANG)
+if (WIN32 AND COMPILER_IS_GCC_OR_CLANG AND NOT COMPILER_IS_ARM_MINGW)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-keep-inline-dllexport")
 endif ()
 
diff '--color=auto' -Naur a/Source/PlatformQt.cmake b/Source/PlatformQt.cmake
--- a/Source/PlatformQt.cmake	2025-07-01 00:16:24.265760354 -0400
+++ b/Source/PlatformQt.cmake	2025-07-01 00:44:33.636545817 -0400
@@ -19,7 +19,11 @@
 # by compiling any static library like WTF or JSC with optimization flags instead and keep
 # debug symbols for the static libraries that implement API.
 cmake_dependent_option(USE_MINIMAL_DEBUG_INFO "Add debug info only for the libraries that implement API" OFF
-    "NOT MINGW;NOT APPLE" ON)
+	"NOT MINGW;NOT APPLE" ON)
+
+if (COMPILER_IS_ARM_MINGW AND USE_MINIMAL_DEBUG_INFO)
+	set(USE_MINIMAL_DEBUG_INFO OFF)
+endif ()
 
 if (USE_MINIMAL_DEBUG_INFO AND CMAKE_BUILD_TYPE STREQUAL "Debug")
     target_compile_options(WTF                PRIVATE -g0 -O1)
@@ -34,10 +38,14 @@
     endif ()
 
     target_compile_options(WebKit            PRIVATE -g1 -O1)
-    target_compile_options(WebKit2           PRIVATE -g1 -O1)
-    if (NOT APPLE)
+    if (TARGET WebKit2)
+        target_compile_options(WebKit2           PRIVATE -g1 -O1)
+    endif ()
+    if (NOT APPLE AND NOT COMPILER_IS_ARM_MINGW)
         target_compile_options(WebKit        PRIVATE -fdebug-types-section)
-        target_compile_options(WebKit2       PRIVATE -fdebug-types-section)
+        if (TARGET WebKit2)
+            target_compile_options(WebKit2       PRIVATE -fdebug-types-section)
+        endif ()
     endif ()
 endif ()
 
@@ -56,7 +64,7 @@
 
 if (FORCE_DEBUG_INFO)
     if (COMPILER_IS_GCC_OR_CLANG)
-        if (NOT APPLE)
+		if (NOT APPLE AND NOT COMPILER_IS_ARM_MINGW)
             target_compile_options(WebKit        PRIVATE -fdebug-types-section)
             target_compile_options(WebKit2       PRIVATE -fdebug-types-section)
         endif ()
